src/video/uikit/SDL_uikitmetalview.m

```diff
@@ -82,12 +82,14 @@ SDL_MetalView UIKit_Metal_CreateView(SDL_VideoDevice *_this, SDL_Window *window)
         SDL_uikitmetalview *metalview;
 
         if (window->flags & SDL_WINDOW_HIGH_PIXEL_DENSITY) {
+#if !TARGET_OS_XR
```

Comment:
Actually, can you move this around the if statement as well? Conceptually we don't want to test the window flags on VisionOS